### PR TITLE
[CSM-426] Serve Swagger doc from `wallet` endpoint

### PR DESCRIPTION
**Pull request by sectore**
How to test?

1. Build everything by running `./scripts/build/cardano-sl.sh`
2. Run wallet api
```
tmux
scripts/launch/demo-with-wallet-api.sh
```
3. Test endpoint w/ `curl`
```
curl --cacert {xxx}/cardano-sl/scripts/tls-files/ca.crt -v https://localhost:8090/docs
```
Or even better: Open https://localhost:8090/docs in your browser (ignore unsafe certificate warning).

You should see something like this:

swagger-wallet

### CODE REVIEW

**Comment by akegalj** [comment]
So it serves json ? I thought it will serve html produced documentation? @sectore

**Comment by Martoon-00** [comment]
Wow, for a difficult thing you ask for

**Comment by akegalj** [comment]
@Martoon-00 
> Wow, for a difficult thing you ask for

how come we have html generated in cardano docs ?

**Comment by sectore** [comment]
@akegalj The goal is to use `SwaggerUI`. The first attempt was to serve a `*.json` file, but this is deprecated now. Please note that this PR is still labeled as `WIP`.

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs [hunk 1]
```diff
@@ -84,3 +96,10 @@ testResetAll = deleteAllKeys >> testReset
     deleteAllKeys = do
         keyNum <- length <$> getSecretKeys
         replicateM_ keyNum $ deleteSecretKey 0
+
+-- | Build Swagger-specification from 'walletApi'.
+swaggerSpec :: Swagger
+swaggerSpec = toCustomSwagger walletApi
+    & info . title       .~ "Cardano SL Wallet Web API"
+    & info . version     .~ toText (showVersion CSL.version)
+    & info . description ?~ "This is an API for Cardano SL wallet."
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: As far as I understand, it's taken from executable. Could you explain why did you remove `host` field?

**Comment by Martoon-00** [comment]
I think it's important to move this instance somewhere to wallet as well
https://github.com/input-output-hk/cardano-sl/pull/1439/files#diff-5f87d064029e0a5c3e34d82a1668aefcR46

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs [hunk 2]
```diff
@@ -14,16 +14,28 @@ module Pos.Wallet.Web.Methods.Misc
...
 import           Universum
 
+import           Control.Lens               ((?~))
+import           Data.Swagger               (Swagger, description, info,
+                                             title, version)
+
+import           Data.Version               (showVersion)
+import qualified Paths_cardano_sl           as CSL
+
+import           Pos.Wallet.Web.Swagger     (toCustomSwagger)
```
Comment (Martoon-00) [inline wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: We're following a rule when imports are grouped like
* Universum
* Modules from standard libraries
* Current project modules

**Comment by sectore** [comment]
> I think it's important to move this instance somewhere to wallet as well

@Martoon-00 Which instances in detail?

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs (hunk 1 above)
Comment (sectore) [reply wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: >  If the host is not included, the host serving the documentation is to be used (including the port).

^ See https://hackage.haskell.org/package/swagger2-2.1.4.1/docs/Data-Swagger.html#g:8

**Comment by Martoon-00** [comment]
Sorry, broken link. This one: https://github.com/input-output-hk/cardano-sl/blob/23f4aa0202971ad35fbbb3ca2a503956bcbfada0/wallet/web-api-swagger/Main.hs#L47

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs (hunk 1 above)
Comment (Martoon-00) [reply wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: Aha, then it makes sense

File: wallet/src/Pos/Wallet/Web/Methods/Misc.hs [hunk 3]
```diff
@@ -68,7 +69,7 @@ applyUpdate = removeNextUpdate >> applyLastUpdate
 ----------------------------------------------------------------------------
 
 syncProgress :: MonadWalletWebMode m => m SyncProgress
-syncProgress = do
+syncProgress =
```
Comment (flyingleafe) [inline wallet/src/Pos/Wallet/Web/Methods/Misc.hs]: ❤️  for small on-the-go refactorings like this
